File: go/fizzbuzz-basic..go
//Implements a basic FizzBuzz solution.
//Prioritizes readability over optimization.
package main

import (
	"fmt"
	"strconv"
)

func main() {
	for i := 1; i <= 100; i++ {

		out := line(i)
		fmt.Println(out)
	}
}

func line(num int) string {
	db3 := divisible(num, 3)
	db5 := divisible(num, 5)

	switch {
	case db3 && db5:
		return "FizzBuzz"
	case db3:
		return "Fizz"
	case db5:
		return "Buzz"
	default:
		return strconv.Itoa(num)
	}
}

func divisible(a int, b int) bool {
	return a%b == 0
}

File: ruby/fizzbuzz-basic.rb
#!/usr/bin/env ruby
# frozen_string_literal: true

# Implements a basic FizzBuzz solution.
# Prioritizes readability over optimization.

# https://rosettacode.org/wiki/FizzBuzz

LOWER_BOUND = 1   #inclusive
UPPER_BOUND = 100

FIZZ_STRING = 'Fizz'
BUZZ_STRING = 'Buzz'

FIZZ_DIVIZOR = 3
BUZZ_DIVIZOR = 5

def main
  LOWER_BOUND.upto(UPPER_BOUND).each do |element|
    line = line(element)
    puts(line)
  end
end

def line(element)
  fizzable = divisible?(element, FIZZ_DIVIZOR)
  buzzable = divisible?(element, BUZZ_DIVIZOR)

  case
  when fizzable && buzzable
    FIZZ_STRING + BUZZ_STRING
  when fizzable
    FIZZ_STRING
  when buzzable
    BUZZ_STRING
  else
    element
  end
end

def divisible?(candidate, divizor)
  candidate % divizor == 0
end

main
